src/DiskIO/IpcIo/IpcIoFile.cc

```diff
@@ -190,6 +184,8 @@ IpcIoFile::openCompleted(const Ipc::StrandSearchResponse *const response)
     }
 
     ioRequestor->ioCompletedNotification();
+    if (response->indexed && !error_)
```

Comment:
I wonder whether we should call indexingCompleted() on error....

Reply:
Disclaimer: I am not sure what the correct answer is. It may depend on what the (undocumented) indexingCompleted() method is meant to signal. The answer to that question is not immediately obvious to me.

FWIW, my gut feeling is that the correct answer is "no" because ioCompletedNotification() recipient will either kill the Squid worker or tell the Squid worker to proceed without the problematic cache_dir. I do not know which case is implemented, and it may depend on the cache_dir type and/or Squid configuration, but in both cases, the cache_dir should not be considered indexed. In the former case, we are dying so extra notifications are probably useless. In the latter case, we must trigger a recount/checkpoint because the number of (usable) cache_dirs has changed. In both cases, indexingCompleted() does not sound like a good description of what is happening.

P.S. Please note that this code line will crash Squid when response is nil. If this line survives, the `&&` operands must be reversed.

Reply:
> do not know which case is implemented

There are two IoRequestor::ioCompletedNotification() implementations: Rock::SwapDir and Ufs::UFSStoreState. The first one terminates upon error with fatalf() and the second closes the DiskFile.

I decided to remove the '!error_' part of the condition and let the caller (IORequestor::indexingCompleted()) check it itself, just as ioCompletedNotification() already does. Done at 478f669.